drivers/timer/ifx_cat1_lp_timer_pdl.c

```diff
@@ -0,0 +1,351 @@
+	lptimer_enable_event(lptimer_dev, true);
+
+	/* passing ticks==1 means "announce the next tick", ticks value of zero (or even negative)
+	 * is legal and treated identically: it simply indicates the kernel would like the next
+	 * tick announcement as soon as possible.
+	 */
+	if (ticks < 1) {
+		ticks = 1;
+	}
+
+	/* Set the delay value for the next wakeup interrupt */
+	lptimer_set_delay(lptimer_dev, ((uint32_t)(ticks)*config->clock_frequency) /
+					       CONFIG_SYS_CLOCK_TICKS_PER_SEC);
+}
+
+uint32_t sys_clock_elapsed(void)
+{
+	const struct device *lptimer_dev =
+		DEVICE_DT_GET(DT_COMPAT_GET_ANY_STATUS_OKAY(DT_DRV_COMPAT));
+	const struct ifx_cat1_lptimer_config *config = lptimer_dev->config;
+	uint32_t lptimer_value;
+	uint64_t ret;
+
+	if (!IS_ENABLED(CONFIG_TICKLESS_KERNEL)) {
+		return 0;
+	}
+
+	k_spinlock_key_t key = k_spin_lock(&lock);
+
+	lptimer_value = Cy_MCWDT_GetCount(config->reg_addr, CY_MCWDT_COUNTER2);
+
+	/* Gives the value of LPTIM counter (ms) since the previous 'announce' */
+	ret = (((uint64_t)(lptimer_value - last_lptimer_value)) * CONFIG_SYS_CLOCK_TICKS_PER_SEC) /
+	      config->clock_frequency;
+
+	k_spin_unlock(&lock, key);
+
+	return (uint32_t)ret;
+}
+
+uint32_t sys_clock_cycle_get_32(void)
+{
+	const struct device *lptimer_dev =
+		DEVICE_DT_GET(DT_COMPAT_GET_ANY_STATUS_OKAY(DT_DRV_COMPAT));
+	const struct ifx_cat1_lptimer_config *config = lptimer_dev->config;
+	uint32_t lp_time;
+	uint64_t ret;
+
+	/* Gives the accumulated count in a number of hw cycles */
+	k_spinlock_key_t key = k_spin_lock(&lock);
+
+	lp_time = Cy_MCWDT_GetCount(config->reg_addr, CY_MCWDT_COUNTER2);
+
+	/* Convert lptim count in a nb of hw cycles with precision */
+	ret = ((uint64_t)lp_time * sys_clock_hw_cycles_per_sec()) / config->clock_frequency;
+
+	k_spin_unlock(&lock, key);
+
+	/* Convert in hw cycles (keeping 32bit value) */
+	return (uint32_t)ret;
+}
+
+static void lptimer_isr(const struct device *dev)
+{
+	const struct ifx_cat1_lptimer_config *config = dev->config;
+	const struct ifx_cat1_lptimer_data *data = dev->data;
+
+	Cy_MCWDT_ClearInterrupt(config->reg_addr, (CY_MCWDT_CTR0 | CY_MCWDT_CTR1 | CY_MCWDT_CTR2));
+
+	/* Clear interrupt mask if set only from cyhal_lptimer_set_delay() function */
+	if (data->clear_int_mask) {
+		Cy_MCWDT_SetInterruptMask(config->reg_addr, 0);
+	}
+
+	if ((data->isr_instruction & LPTIMER_ISR_CALL_USER_CB_MASK) != 0) {
+		/* announce the elapsed time in ms */
+		uint32_t lptimer_value = Cy_MCWDT_GetCount(config->reg_addr, CY_MCWDT_COUNTER2);
+		int32_t delta_ticks = ((uint64_t)(lptimer_value - last_lptimer_value) *
+				       CONFIG_SYS_CLOCK_TICKS_PER_SEC) /
+				      config->clock_frequency;
+		sys_clock_announce(IS_ENABLED(CONFIG_TICKLESS_KERNEL) ? delta_ticks
+								      : (delta_ticks > 0));
+		last_lptimer_value = lptimer_value;
+	}
+}
+
+static int lptimer_init(const struct device *dev)
+{
+#define LPTIMER_CTRL (CY_MCWDT_CTR0 | CY_MCWDT_CTR1 | CY_MCWDT_CTR2)
```

Comment:
The #define in the function body looks a bit odd, I'd shift it out or make it a const value if the only use is in this function.